### PR TITLE
[SE-3182] Remove certbot & acme from Pipfile.

### PR DESCRIPTION
**Pull request by mtyaka**
This removes certbot from pipenv and uses the system-provided certbot instead. This change is required because we were running into compatibility issues between the version of certbot that was pinned in the pipenv (used for requesting new certificates) and the system-level certbot installation (used for renewing certificates).

See [SE-3182](https://tasks.opencraft.com/browse/SE-3182) for more info.

**Test instructions**

This change was deployed to `cert-manager-stage.net.opencraft.hosting` together with https://github.com/open-craft/ansible-playbooks/pull/181.

Create a new instance on https://stage.manage.opencraft.com/ and spawn a new appserver. Verify that a certificate for your instance gets provisioned successfully (note that Ocim stage uses the staging letsencrypt environment, so you will see an "unknown issuer" warning in your browsers).

**Reviewers**:

- [ ] @Kelketek
- [ ] @lgp171188

### CODE REVIEW

**Comment by Kelketek** [comment]
@mtyaka Visiting https://stage.manage.opencraft.hosting/ errors out for me-- and not just on the cert level:

Screen Shot 2020-09-10 at 9 42 28 AM

I'm not sure if this is related to your changes, but it's being looked at now.

**Comment by Kelketek** [comment]
Ah, I see. I think you meant to link me to https://stage.manage.opencraft.com/ . The link you gave me was to a specific instance. I'm going to try provisioning something new now.

**Comment by mtyaka** [comment]
Yes, I meant https://stage.manage.opencraft.com/ 🤦 Sorry about that. I fixed the link in the description.

**Comment by lgp171188** [comment]
@mtyaka, this is a great change because `cert-manager` only invokes the `certbot` as a command-line tool and doesn't use it as a library at all. If our playbooks can ensure that `certbot` installed and the corresponding command is available, this need not be a dependency any more.

**Comment by mtyaka** [comment]
> this is a great change because `cert-manager` only invokes the `certbot` as a command-line tool and doesn't use it as a library at all. If our playbooks can ensure that `certbot` installed and the corresponding command is available, this need not be a dependency any more.

Yes, I was happy to find out cert-manager only uses `certbot` through the CLI so the changes required here were minimal :)
 
> With the automatic snap package upgrades (which cannot be blocked or delayed more than a certain period), we have to keep any eye out for the `certbot` CLI command changes that might break `cert-manager`.

Yes, that's something we'll have to be aware of.

File: Pipfile [hunk 1]
```diff
@@ -6,9 +6,7 @@ name = "pypi"
 [packages]
 python-consul = "~= 1.1"
 pyopenssl = "~= 18.0"
-certbot = "==1.1.0"
-acme = "==1.1.0"
-mock = "==3.0.5"
+cryptography = "~= 2.9"
```
Comment (lgp171188) [inline Pipfile:9]: @mtyaka, is there a reason why this is now an explicit dependency? I can't find any direct usage of this library at all.

**Comment by lgp171188** [comment]
@mtyaka, while the certificate provisioning via the `cert-manager` appears to work, I am kinda surprised that it does because the `certbot` command is not in `$PATH` and the command invocation [here](https://github.com/open-craft/cert-manager/blob/master/cert_manager/certbot.py#L20-L31) doesn't provide the absolute path to the `certbot` command.

Even if there is a reason for why this is working, should we make it explicit?

CC @Kelketek

File: Pipfile (hunk 1 above)
Comment (mtyaka) [reply Pipfile:9]: This is a transitive dependency, but without manually pinning it in the Pipfile, `pipenv install` will install the latest version which is not compatible with python 3.5.

**Comment by mtyaka** [comment]
@lgp171188 

> @mtyaka, while the certificate provisioning via the cert-manager appears to work, I am kinda surprised that it does because the certbot command is not in $PATH and the command invocation here doesn't provide the absolute path to the certbot command.
> Even if there is a reason for why this is working, should we make it explicit?

Good point, I'll add a configuration variable `CERTBOT_PATH` and I'll make it default to `/snap/bin/certbot`.

**Review by lgp171188 — APPROVED** [review]
:+1: 
- [x] I tested this on stage and the certificate provisioning worked okay, which means the removal of these dependencies didn't break anything.
- [x] I read through the code

**Comment by Kelketek** [comment]
Just wanted to prod this PR and confirm that it's good to go from my end, in case that wasn't clear. 👍

**Pull request merged**